client/src/components/Post/PostList.tsx

```diff
@@ -1,54 +1,42 @@
 import { useDispatch, useSelector } from 'react-redux';
 import allAction from '../../redux/actions';
-import { useEffect, useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { Post } from '../../types/Post'
 import PostItem from './PostItem'
 
 const PostList = () => {
-  const fetchedPosts = useSelector((state:any) => state.postList);
-  const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(allAction.loadPostList());
-  }, []);
-
   const [pageCount, setPageCount] = useState(0)
-  const [fetching, setFetching] = useState(false)
-  const [posts, setPosts] = useState<Post[]>([])
+  const [fetchingPost, setFetchingPost] = useState(false)
 
   const MAXIMUM_PAGES = 10
 
-  const handleScroll = () => {
-    const scrollHeight = document.documentElement.scrollHeight
-    const scrollTop = document.documentElement.scrollTop
-    const clientHeight = document.documentElement.clientHeight
+  const fetchedPosts = useSelector((state:any) => state.postList);
+  const dispatch = useDispatch();
 
+  const handleScroll = () => {
+    const {scrollHeight, scrollTop, clientHeight } = document.documentElement;
     if (scrollTop + clientHeight + 100 >= scrollHeight) {
-      fetchMoreData()
+      setFetchingPost(true)
     }
   }
 
-  window.addEventListener('scroll', handleScroll)
-  useEffect(() => {
-    return () => {
-      window.removeEventListener('scroll', handleScroll)
-    }
-  }, [])
-
-  const appendData = () => {
-    setFetching(true)
-    setPosts(fetchedPosts)
+  const fetchArticle = useCallback(()=> {
+    dispatch(allAction.loadPostList());
     setPageCount(pageCount + 1)
-    setFetching(false)
-  }
+    setFetchingPost(false)
+  },[pageCount])
 
+  useEffect(() => {
+    handleScroll()
+    setFetchingPost(true)
+    window.addEventListener('scroll', handleScroll)
```

Comment:
```suggestion
    fetch()
    window.addEventListener('scroll', handleScroll)
```
정도로만 하고 fetching에 대한 데이터도 store에서 관리하는게 좋겠네요!

Reply:
reducer에 fetch요청하면
if false, `fetch = true`하고 reqeust한 후 응답 받았을 때 `fetch = false`
else return

하는 방식으로요